client/src/main/java/io/pravega/client/netty/impl/ConnectionPoolImpl.java

```diff
@@ -204,11 +204,11 @@ public int getActiveChannelCount() {
         } catch (Exception e) {
             connectionComplete.completeExceptionally(new ConnectionFailedException(e));
         }
-
-        final CompletableFuture<Void> channelRegisteredFuture = new CompletableFuture<>(); //to track channel registration.
-        handler.completeWhenRegistered(channelRegisteredFuture);
-
-        return CompletableFuture.allOf(connectionComplete, channelRegisteredFuture);
+        return connectionComplete.thenCompose(v ->  {
```

Comment:
+1, we should not wait for the `channelRegistredFuture` to be completed (either exceptionally or successfully) if the CompletableFuture `connectionComplete` completes exceptionally.